src/main/java/org/jenkinsci/plugins/stashNotifier/StashNotifier.java

```diff
@@ -181,105 +184,140 @@ public BuildStepMonitor getRequiredMonitorService() {
             boolean considerUnstableAsSuccess,
             JenkinsLocationConfiguration globalConfig
     ) {
+        this.globalConfig = globalConfig;
+        setStashServerBaseUrl(stashServerBaseUrl);
+        setCredentialsId(credentialsId);
+        setIgnoreUnverifiedSSLPeer(ignoreUnverifiedSSLPeer);
+        setCommitSha1(commitSha1);
+        setBuildStatus(buildStatus);
+        setBuildName(buildName);
+        setIncludeBuildNumberInKey(includeBuildNumberInKey);
+        setProjectKey(projectKey);
+        setPrependParentProjectKey(prependParentProjectKey);
+        setDisableInprogressNotification(disableInprogressNotification);
+        setConsiderUnstableAsSuccess(considerUnstableAsSuccess);
+    }
 
-        this.stashServerBaseUrl = stashServerBaseUrl != null && stashServerBaseUrl.endsWith("/")
-                ? stashServerBaseUrl.substring(0, stashServerBaseUrl.length() - 1)
-                : stashServerBaseUrl;
-        this.credentialsId = credentialsId;
-        this.ignoreUnverifiedSSLPeer = ignoreUnverifiedSSLPeer;
-        this.commitSha1 = commitSha1;
-
-        StashBuildState overwrittenBuildState = null;
-        try {
-            overwrittenBuildState = StashBuildState.valueOf(buildStatus);
-        } catch (Exception e) {
-            // ignore unknown or null values
-        }
-        this.buildStatus = overwrittenBuildState;
-
-        this.buildName = buildName;
-        this.includeBuildNumberInKey = includeBuildNumberInKey;
-        this.projectKey = projectKey;
-        this.prependParentProjectKey = prependParentProjectKey;
-        this.disableInprogressNotification = disableInprogressNotification;
-        this.considerUnstableAsSuccess = considerUnstableAsSuccess;
+    StashNotifier(
```

Comment:
I am thinking of the number of constructors can be reduced even more? Is this one really required?

Reply:
I'm not sure about that, and neither am I that the test cases existing as well as the ones I can do locally cover all potential use cases. So I've kept it to be on the safe side.

Reply:
But this is a new one and another one has been removed.

In other plugins it's often handled the way to keep all old constructors for backward compatibility and annotate them with `@Deprecated`. That's something the maintainers should decide.

Reply:
Oh you're right. Then I'm not totally sure, what the constructor was meant for. I think it's just the old `DataBoundConstructor` without all of the optional parameters. So I could just move the logic in the no-argument constructor.

Reply:
Was some time out of the country and noticed, that I did not push the change before. Done that now.